### PR TITLE
WIP: Add saml login.

### PR DESCRIPTION
**Pull request by sashax**

### CODE REVIEW

**Comment by sashax** [comment]
Current setup assumes identity-rp is running locally, and that identity-dip is running on upaya-idp-dev.18f.gov

TODO:
 - [ ] Fail page
 - [ ] Tests!
 - [ ] Test on server

File: app.rb [hunk 1]
```diff
@@ -16,17 +21,52 @@ def auth_server_uri
...
+      redirect to('/success')
+    end
+  end
+
+  private
+
+  def saml_settings
+    settings = OneLogin::RubySaml::Settings.new(YAML.load_file 'config/saml_settings.yml')
+    settings.certificate = File.read('config/server.crt')
+    settings.private_key = File.read('config/server.key')
```
Comment (brendansudol) [inline app.rb]: where can i get this server.key?

**Comment by brendansudol** [comment]
@sashax, I'd be happy to merge this PR and add tests + fail page in future PRs -- thoughts?

**Comment by sashax** [comment]
Sure. @ertzeid is working on moving everything to Rails in parallel, but if it's not disruptive to her, I say go for it. 

Note that I'm upgrading the cert stuff as well (to be 2048 bit and better named), but that's orthogonal to this.

**Comment by brendansudol** [comment]
👍

**Pull request merged**